### ISSUE TITLE
Feature: Plausible Analytics Integration (Opt-in Only)

### ISSUE THREAD

**Issue by lreading** [OWASP/threat-dragon#1712]
**Describe what problem your feature request solves**:  

I intend to leave this issue open for some time for feedback before deciding if we should move forward with it. If you have opinions on this, _please_ comment. I just ask that you read the issue in full first. 😄 

### Why?
Analytics are potentially useful for at least 2 audiences:
1. Organizations who want to understand how Threat Dragon is being used
2. Threat Dragon maintainers / community - a public and open dashboard helps maintainers understand what features are being used and _sometimes_ how. This helps us prioritize bug fixes and new features based on what the community is actually using.

### What Would You Track?
Let's start with what we wouldn't track: any kind of user-supplied data, including organization name, repository name, threat model details of any kind, github username, etc. **We are not interested in identifying users.**

Here are some actions that I think would be helpful to track:
- docs link clicked (potentially correlate the current view)
- OWASP links click
- language selection changed - do not record the language
- logged in via [github|localsession\ - do not record the username or other identifiable data
- open existing TM - no names or metadata
- create new model
- etc

I don't want to try to iterate every possible action at the moment. But it's safe to assume that I'd include most navigation and significant page actions, so long as the event doesn't contain identifiable data.

### Why Palusible?
[Plausible](https://github.com/plausible/analytics) is open source, privacy focused, and affordable. I'd recommend reading [Plausible.io's data policy](https://plausible.io/data-policy) if you are not familiar with them. They explain everything they collect as well as how it's used.

Plausible provides the option to have your analytics dashboard be public. Threat Dragon's analytics will be public. This is not only in the spirit of open source, but also a way for people to verify what is tracked on the demo site.

### Transparency
When analytics are enabled, a link should be present to a new "analytics" view. That view would include the url/identifier of the Plausible instance, what data is collected, and a full list of all events tracked. 

The list of events to track will be kept as an enum - that allows any user to view the event names that are deployed at any point in time.

**Describe the solution you'd like**:  

This config would be exposed by the /api/config endpoint:
```bash
PLAUSIBLE_ENABLED=true # default false
PLAUSIBLE_URL=some.site
PLAUSIBLE_ID=12345 # I'm assuming
```

in td.vue, only load Plausible if the config says it's enabled. 
If loaded, a link to the analytics page must be present and conspicuous
Include a link on the analytics page for Threat Dragon's public dashboards

Configure www.threatdragon.com to use a hosted Plausibe instance and share all dashboards publicly
For the above, I would be willing to sponsor the cost for the hosted version. Alternatively, I could also self-host it, keeping it all on the threatdragon.com domain. I just can't make any promises on uptime (my setup's pretty good, but not 5 9's 😆)

**Declaration**:  

By submitting this issue you have:  

- [x] read the [contribution guide](../blob/main/contributing.md) and agree to the [Code of Conduct](../blob/main/code_of_conduct.md)
- [x] not used agentic or generative AI in creating this feature request

**Additional context**:

**Comment by lreading** [comment]
Docs: 
https://www.threatdragon.com/docs/trust/analytics.html
https://www.threatdragon.com/docs/configure/analytics.html

When analytics are enabled, Threat Dragon shows an animated chart icon in the header, linking to the configured Plausible dashboard.

https://www.threatdragon.com/ is configured to use a self-hosted instance of Plausible.

Dashboards:
- [Main dashboard](https://analytics.threatdragon.com/share/threatdragon.com?auth=aCV0lHHVgvub06ga4z9by)
- [Application areas](https://analytics.threatdragon.com/share/threatdragon.com?auth=V0ELzy6vHE2eEQKcHV3SP)
- [Model workflows](https://analytics.threatdragon.com/share/threatdragon.com?auth=-2sskTQwJVjCbszgx6izf)
- [Editing session duration](https://analytics.threatdragon.com/share/threatdragon.com?auth=oqZgZ-ld4dQ-BsJMgdWho)
- [Language adoption](https://analytics.threatdragon.com/share/threatdragon.com?auth=T5wJ6xCqvjKfw0bDt-AFc)
- [Diagram workflows](https://analytics.threatdragon.com/share/threatdragon.com?auth=qY4zbuxFc6Cw_cVTL8RGN)
- [Threat workflows](https://analytics.threatdragon.com/share/threatdragon.com?auth=a_l01Z1B6a-JX5t5mrfEw)
- [Threat engine](https://analytics.threatdragon.com/share/threatdragon.com?auth=2kwgJvoGIX-ECkAlpUlUq)
- [Exports](https://analytics.threatdragon.com/share/threatdragon.com?auth=U0hxcACC1QuyeCp6mMY_L)
